src/java/io/bazel/rulesscala/test_discovery/FilteredRunnerBuilder.scala

```diff
@@ -19,9 +21,7 @@ class FilteredRunnerBuilder(builder: RunnerBuilder) extends RunnerBuilder {
         runner match {
           // Regular JUnit tests.
           case _: BlockJUnit4ClassRunner => new FilteredClassRunner(testClass, pattern)
-          // TODO: handle specs2
-          case _: Suite => runner
-          case _ => runner
+          case _ => runnerFilter(runner, testClass, pattern)
```

Comment:
IIUC, this is only needed if a test class type is assigned an incorrect rule in bazel.
i.e. Junit test assigned a specs2_test target.
Is that correct?

Reply:
Actually, this is needed due to the inheritance, right?

Reply:
Yes, inheritors can provide a different filter implementation.